mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/TransactionalDeclarationEnricher.java

```diff
@@ -81,32 +88,45 @@ public void enrich(ExtensionLoadingContext extensionLoadingContext) {
 
         @Override
         protected void onSource(SourceDeclaration declaration) {
-          addTxParameter(TRANSACTIONAL_ACTION_PARAMETER_NAME, sourceTransactionalActionType, NONE,
+          addTxParameter(TRANSACTIONAL_ACTION_PARAMETER_NAME, sourceTransactionalActionType, sdkSourceTransactionalActionType,
+                         NONE, org.mule.sdk.api.tx.SourceTransactionalAction.NONE,
                          SOURCE_TRANSACTIONAL_ACTION_PARAMETER_DESCRIPTION, declaration, new TransactionalActionModelProperty());
-          addTxParameter(TRANSACTIONAL_TYPE_PARAMETER_NAME, transactionType, LOCAL, TRANSACTION_TYPE_PARAMETER_DESCRIPTION,
+          addTxParameter(TRANSACTIONAL_TYPE_PARAMETER_NAME, null, transactionType, null, LOCAL,
+                         TRANSACTION_TYPE_PARAMETER_DESCRIPTION,
                          declaration,
                          new TransactionalTypeModelProperty());
         }
 
         @Override
         protected void onOperation(OperationDeclaration declaration) {
-          addTxParameter(TRANSACTIONAL_ACTION_PARAMETER_NAME, operationTransactionalActionType, JOIN_IF_POSSIBLE,
+          addTxParameter(TRANSACTIONAL_ACTION_PARAMETER_NAME, operationTransactionalActionType,
+                         sdkOperationTransactionalActionType, JOIN_IF_POSSIBLE,
+                         org.mule.sdk.api.tx.OperationTransactionalAction.JOIN_IF_POSSIBLE,
                          OPERATION_TRANSACTIONAL_ACTION_PARAMETER_DESCRIPTION, declaration,
                          new TransactionalActionModelProperty());
         }
       }.walk(extensionLoadingContext.getExtensionDeclarer().getDeclaration());
     }
 
-    private void addTxParameter(String parameterName, MetadataType metadataType, Object defaultValue, String description,
+    private void addTxParameter(String parameterName, MetadataType metadataType, MetadataType sdkMetadataType,
+                                Object defaultValue, Object sdkDefaultValue, String description,
                                 ExecutableComponentDeclaration<?> declaration, ModelProperty modelProperty) {
       if (declaration.isTransactional()) {
         Optional<ParameterDeclaration> parameterDeclaration = isPresent(declaration, metadataType);
-        if (parameterDeclaration.isPresent()) {
+        Optional<ParameterDeclaration> sdkParameterDeclaration = isPresent(declaration, sdkMetadataType);
```

Comment:
It seems weird to me that an Enricher in extensions-api needs to be this aware of java types. Maybe there is a part that can go in a Java specific enricher, and a more enricher here in extensions-api?

Reply:
Agreed but I think this should be a separate issue, as before my change the enricher was already aware of the java metadata type.